src/util/db/dbconnectionpool.cpp

```diff
@@ -71,16 +71,20 @@ DbConnectionPool::DbConnectionPool(
       m_connectionCounter(0) {
 }
 
-DbConnectionPool::ThreadLocalScope::ThreadLocalScope(
+DbConnectionPool::ThreadLocalScoped::ThreadLocalScoped(
```

Comment:
Well, in that sense it effectively is a thread-local database connection. I will think about how the design could be aligned with that of the low-level DbConnection class. The low-level DbConnection class will only be used behind the scenes for resource management and should not appear application code. Maybe I can hide it entirely.

Good input, @daschuer. It's always helpful to look at things from a different angle! And it's often the "stupid" questions that open your eyes ;) But sadly not everyone understands those principles of a peer review, exhausting discussions in daily business :(